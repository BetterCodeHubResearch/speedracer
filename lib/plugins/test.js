const path = require('path')

const { createDir } = require('../.internal/util')

export default (options) => (reports, file) => {
  reports.forEach(report => {
    const dirname = path.join(path.dirname(file), options.snapshotsDirname)
    const filename = path.join(dirname, `${report.metadata.id}.speedracer`)

    return loadReport(filename).then(
      oldReport => {
        process.exit(0)
      },
      () => createDir(dirname).then(
        () => writeFile(filename, race.report.serialize())
      )
    )
  })
}
