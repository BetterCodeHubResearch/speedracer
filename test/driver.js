// Packages
import delay from 'delay'
import sinon from 'sinon'
import test from 'ava'

// Ours
import { Page, Tracing } from './helpers/mock-chrome-remote'
import createDriver from '../lib/driver'

test.afterEach(t => {
  Object.keys(Page).concat(Object.keys(Tracing))
    .map(k => Page[k] || Tracing[k])
    .filter(f => typeof f === 'function')
    .forEach(f => f.reset())
})

test('load a file', async t => {
  const driver = await createDriver({ port: 3000 })
  await driver.loadFile('test/fixtures/fast.js')
  t.snapshot(Page.navigate.args)
  driver.close()
})

test('trace a file', async t => {
  const handler = sinon.spy()
  const driver = await createDriver({ port: 3000 })
  driver.on('status', handler)
  await driver.loadFile('test/fixtures/fast.js')
  await driver.startTracing()
  await driver.stopTracing()
  t.true(Tracing.start.calledOnce)
  t.snapshot(handler.args)
  driver.close()
})

test('collect events', async t => {
  const driver = await createDriver({ port: 3000 })
  await driver.loadFile('test/fixtures/fast.js')
  await driver.startTracing()
  await delay(0)
  await driver.stopTracing()
  t.true(Tracing.callbacks.dataCollected.calledOnce)
  t.snapshot(Tracing.callbacks.dataCollected.args)
  driver.close()
})
