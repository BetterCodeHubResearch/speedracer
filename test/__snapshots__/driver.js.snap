// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`collect events 1`] = `
Array [
  Array [
    Object {
      "value": Array [],
    },
  ],
]
`;

exports[`load a file 1`] = `
Array [
  Array [
    Object {
      "url": "http://localhost:3000/test/fixtures/fast",
    },
  ],
]
`;

exports[`trace a file 1`] = `
Array [
  Array [
    Object {
      "bufferUsage": 0.01,
      "collectedEvents": 0,
    },
  ],
  Array [
    Object {
      "bufferUsage": 0.01,
      "collectedEvents": 0,
    },
  ],
]
`;
